client/engine/chainservice/adjudicator/reclaim.go

```diff
@@ -0,0 +1,59 @@
+package NitroAdjudicator
+
+import (
+	"fmt"
+
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+)
+
+// computeReclaimEffects mirrors on chain code.
+// It computes side effects for the reclaim function. Returns updated allocations for the source, computed by finding the guarantee in the source for the target, and moving money out of the guarantee and back into the ledger channel as regular allocations for the participants.
+func computeReclaimEffects(sourceAllocations []outcome.Allocation, targetAllocations []outcome.Allocation, indexOfTargetInSource uint) ([]outcome.Allocation, error) {
```

Comment:
It seems that the amount a source channel allocates to a target channel is never checked. Is this a concern for underfunded channels?

Reply:
Good catch! I think this check is missing from the satp paper (which we are using as a spec for this work). 

I think we need to add that check!